hfs/preprocessing.py

```diff
@@ -123,28 +123,44 @@ def get_hierarchy(self):
             In this case the hierarchy graph has not been updated yet.
         """
         if self.is_fitted_:
-            output_hierarchy = self._hierarchy
+            output_hierarchy = self._hierarchy_graph
             output_hierarchy.remove_node("ROOT")
-            return nx.to_numpy_array(self._hierarchy)
+            return nx.to_numpy_array(self._hierarchy_graph)
         else:
             raise RuntimeError("Instance has not been fitted.")
 
     def _extend_dag(self):
         """Adds missing nodes to the hierarchy graph.
 
-        For columns that no have a corresponding node in the hierarchy a
+        For columns that don't have a corresponding node in the hierarchy a
         node is added right under the "ROOT" node.
+        We then update the columns mapping to include the new nodes.
+        If a node in the hierarchy has a name conflict with a column in the
+        dataset we add a node with the next available id.
         """
-        max = len(self._hierarchy.nodes) - 1
-        for x in range(len(self._columns)):
-            if self._columns[x] == -1:
-                if x in self._hierarchy.nodes:
-                    self._hierarchy.add_edge("ROOT", max)
-                    self._columns[x] = max
-                    max += 1
+        # -1 to account for the "ROOT" node
+        next_available_node_id = len(self._hierarchy_graph.nodes) - 1
+        columns_without_node = []
+
+        for column_index, column_mapping in enumerate(self._columns):
+            if column_mapping == -1:  # no corresponding node yet
+                columns_without_node.append(column_index)
+                if column_index in self._hierarchy_graph.nodes:
```

Comment:
Can you add a comment between lines 147 and 148 to shortly touch upon the rationale for this inner if-statement?

Reply:
Isn't this is explained by the comment that is in the next line, within the inner-most if statement? What info do you think is missing here?

Reply:
You are right, maybe we'd rather take a note that some sort of explanation should be included in the full documentation.